client/pkg/batch/batch_controller.go

```diff
@@ -60,13 +60,18 @@ func NewController[T any](maxBatchSize int, finisher FinisherFunc[T], bestBatchO
 // It returns nil error if everything goes well, otherwise a non-nil error which means we should stop the service.
 // It's guaranteed that if this function failed after collecting some requests, then these requests will be cancelled
 // when the function returns, so the caller don't need to clear them manually.
+// `tokenCh` is an optional parameter:
```

Comment:
Shall we use something like
```
FetchPendingRequests(ctx context.Context, requestCh <-chan T, maxBatchWaitInterval time.Duration, tokenCh ...chan struct{}) (errRet error) {
```

Reply:
It looks good to me, but I prefer keeping the current way to force the caller to decide whether `tokenCh` is needed by explicitly passing it in.